### ISSUE TITLE
[Bug]: Can rename file/folder in group folder without delete permission

### ISSUE THREAD

**Issue by luka-nextcloud** [nextcloud/groupfolders#3276]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Users can rename a file/folder when the right of delete file is denied in advanced authorization. Happens on 29 and master.

### Steps to reproduce

1. Create a group folder "GroupFolder1"
2. Create a group "Group1"
3. Assign user "UserA" to group "Group1"
4. Assign group "Group1" to "GroupFolder1"
5. Enable "advanced permissions" for group folder "GroupFolder1" and select "Group1"
6. Login as "UserA"
7. Access files
8. Open sharing tab of folder "GroupFolder1"
9. Add advanced permission for group "Group1". Allow read, write, create, share. Deny delete
10. Access folder "GroupFolder1"
11. Create any file/folder
12. Rename created file/folder

### Expected behavior

File/Folder cannot be renamed without delete permission.

### Installation method

None

### Nextcloud Server version

29

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by provokateurin** [comment]
Hm I'd say this is intended. Delete means non-recoverable changes while renaming a file or folder can be reverted easily.
I'd say this can be fixed with https://github.com/nextcloud/groupfolders/issues/1646 by just adding it to the ACL options.
I'm not sure if it should be the same as the "Move" ACL or a separate "Rename" ACL.
@joshtrichards what do you think?

**Comment by niziak** [comment]
Our whole concept of ACL plan is now broken. We don't want to enable DELETE permission because it is too dangerous. This is definitely a **bug**. Rename without delete ACL was default behaviour for years. After years of working feature there is no place for discussion how it should work. It should works as before!

**Comment by xBlazeFirex** [comment]
We have the same error in our company.
The User must have the "delete" permission in the group that he can change/rename folder names...
That's not good cause not everybody should have the delete-permissions....

**Comment by niziak** [comment]
I think this it is not `groupfolders` related issue. There are more issues about renaming (shared folders) after upgrade NC from 29 to 30.

**Comment by andriusr** [comment]
Feature to disable move/rename of the folder was requested by many users of Nextcloud. There is still open issue on Github that has links to user forum:
https://github.com/nextcloud/server/issues/40881
Still, there are conflicting reports above If removing DELETE permission would result in inability to rename/move folder. Could somebody provide feedback how it works on different Nexcloud versions, in particular 29.x and 30.x?

**Comment by brunodegoyrans** [comment]
I totally agree, a user may have the right to RENAME but no right to DELETE.

DELETE is very very dangerous. I have lost entire projects in the past because a user decided to clean its hard drive. A broad deletion can also come from a malicious employee.

Users shall have the right to rename if they have the right to create. Otherwise, you will have tons of "new file/folder" or "file/folder without title" named files/folders.

It was working like that before I migrate to nexcloud 30. I spent days to find out, thinking it was my config. I'am so happy to find this issue on github. I hope NC will fix it soon.

Today, I have to rename myself all the files. I have several requests each days. I'am the boss, I should do bossing.

**Comment by niziak** [comment]
Problematic change was introduced by https://github.com/nextcloud/server/pull/47185
With this logic of handling rename in NC - treat as copy + delete so `Permission.DELETE` is required, `GroupFolders` app needs to add own "rename" file action to implement rename without `Permission.DELETE`.

**Comment by daFlo** [comment]
@luka-nextcloud 

So with https://github.com/nextcloud/server/issues/47104 this change was rolled out without a proper solution for installations having a >20 of nonIT-users. You absolutely can´t grant those users delete- or move-rights or you will have a big mess within a short time. Would be nice to know if there is any perspective or we will be stuck to v29.

**Comment by yetopen-com** [comment]
@niziak did you find a workaround? 
thank's

**Comment by kovacsakos14** [comment]
Is there any update on this?
@niziak Were you able to create the add-on?
